openmls/src/treesync/diff.rs

```diff
@@ -296,21 +297,47 @@ impl<'a> TreeSyncDiff<'a> {
         ciphersuite: Ciphersuite,
         group_id: GroupId,
         leaf_index: LeafNodeIndex,
-    ) -> Result<UpdatePathResult, LibraryError> {
-        debug_assert!(
-            self.leaf(leaf_index).is_some(),
-            "Tree diff is missing own leaf"
-        );
+        leaf_node_params: SimpleLeafNodeParams,
+    ) -> Result<UpdatePathResult, TreeSyncAddLeaf> {
+        // We temporarily add a placeholder leaf node to the tree, because it
+        // might be required to make the tree grow to the right size. If we
+        // don't do that, calculating the direct path might fail. It's important
+        // to not do anything with the value of that leaf until it has been
+        // replaced.
+        let leaf_node = LeafNode::new_placeholder();
+
+        // In case of an External Commit, where the leaf is not replaced and the
+        // tree is full, we have to grow the tree first. In all other cases we
+        // just replace an existing leaf.
+        if leaf_index.u32() >= self.leaf_count() {
+            self.add_leaf(leaf_node)?;
```

Comment:
Should we have a `debug_assert!` here that we are processing an external commit (and one in the else branch  that we are processing a regular commit, respectively)?

Reply:
I refactored this anyway now, so I thing the comment no longer valid